src/Take.Elephant.Sql/DbDataReaderAsyncEnumerable.cs

```diff
@@ -39,7 +39,8 @@ public virtual async Task<IAsyncEnumerator<T>> GetEnumeratorAsync(CancellationTo
 
         public virtual IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken)
         {
-            return GetEnumeratorAsync(cancellationToken).Result;
+            // TODO: This is a blocking operation in a async context. How may we fix this?
```

Comment:
The implementation is flawed. The `GetEnumeratorAsync` method above should not exist. Instead, this method should prepare all the dependencies and send them to the enumerator. `DbDataReaderAsyncEnumerator.MoveNextAsync` implementation will be a state machine; When invoked on its initial state, it will create and `OpenAsync` the connection and then `ExecuteReaderAsync`. Following invocations would use the current implementation.